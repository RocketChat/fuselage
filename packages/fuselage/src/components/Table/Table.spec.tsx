import React from 'react';
import ReactDOM from 'react-dom';

import { Table } from '../..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Table />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Head', () => {
  it('renders without crashing', () => {
    const table = document.createElement('table');
    ReactDOM.render(<Table.Head />, table);
    ReactDOM.unmountComponentAtNode(table);
  });
});

describe('Body', () => {
  it('renders without crashing', () => {
    const table = document.createElement('table');
    ReactDOM.render(<Table.Body />, table);
    ReactDOM.unmountComponentAtNode(table);
  });
});

describe('Foot', () => {
  it('renders without crashing', () => {
    const table = document.createElement('table');
    ReactDOM.render(<Table.Foot />, table);
    ReactDOM.unmountComponentAtNode(table);
  });
});

describe('Row', () => {
  it('renders without crashing', () => {
    const tbody = document.createElement('tbody');
    ReactDOM.render(<Table.Row />, tbody);
    ReactDOM.unmountComponentAtNode(tbody);
  });
});

describe('Cell', () => {
  it('renders without crashing', () => {
    const tr = document.createElement('tr');
    ReactDOM.render(<Table.Cell />, tr);
    ReactDOM.unmountComponentAtNode(tr);
  });
});
