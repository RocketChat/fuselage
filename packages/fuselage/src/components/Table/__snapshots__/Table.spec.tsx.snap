// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Table Rendering] renders Default without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-table__wrapper"
    >
      <table
        class="rcx-box rcx-box--full rcx-table"
      >
        <thead
          class="rcx-box rcx-box--full rcx-table__head"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell"
            >
              Dessert (100g serving)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Calories
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Fat (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Carbs (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Protein (g)
            </th>
          </tr>
        </thead>
        <tbody
          class="rcx-box rcx-box--full rcx-table__body"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row--action rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Ice cream sandwich
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              237
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              9
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              37
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4.3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`[Table Rendering] renders Fixed without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-table__wrapper"
    >
      <table
        class="rcx-box rcx-box--full rcx-table--fixed rcx-table"
      >
        <thead
          class="rcx-box rcx-box--full rcx-table__head"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell"
            >
              Dessert (100g serving)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Calories
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Fat (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Carbs (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Protein (g)
            </th>
          </tr>
        </thead>
        <tbody
          class="rcx-box rcx-box--full rcx-table__body"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Ice cream sandwich
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              237
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              9
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              37
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4.3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`[Table Rendering] renders Striped without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-table__wrapper"
    >
      <table
        class="rcx-box rcx-box--full rcx-table--striped rcx-table--sticky rcx-table--fixed rcx-table"
      >
        <thead
          class="rcx-box rcx-box--full rcx-table__head"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell"
            >
              Dessert (100g serving)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Calories
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Fat (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Carbs (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Protein (g)
            </th>
          </tr>
        </thead>
        <tbody
          class="rcx-box rcx-box--full rcx-table__body"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Ice cream sandwich
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              237
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              9
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              37
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4.3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</body>
`;

exports[`[Table Rendering] renders WithSelection without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-table__wrapper"
    >
      <table
        class="rcx-box rcx-box--full rcx-table"
      >
        <thead
          class="rcx-box rcx-box--full rcx-table__head"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select all"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell"
            >
              Dessert (100g serving)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Calories
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Fat (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Carbs (g)
            </th>
            <th
              class="rcx-box rcx-box--full rcx-table__cell--header rcx-table__cell--align-end rcx-table__cell"
            >
              Protein (g)
            </th>
          </tr>
        </thead>
        <tbody
          class="rcx-box rcx-box--full rcx-table__body"
        >
          <tr
            class="rcx-box rcx-box--full rcx-table__row--selected rcx-table__row"
          >
            <td
              class="rcx-box rcx-box--full rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select frozen yoghurt"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </td>
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row--selected rcx-table__row"
          >
            <td
              class="rcx-box rcx-box--full rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select frozen yoghurt"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </td>
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row"
          >
            <td
              class="rcx-box rcx-box--full rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select frozen yoghurt"
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </td>
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row--selected rcx-table__row"
          >
            <td
              class="rcx-box rcx-box--full rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select frozen yoghurt"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </td>
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
          <tr
            class="rcx-box rcx-box--full rcx-table__row--selected rcx-table__row"
          >
            <td
              class="rcx-box rcx-box--full rcx-table__cell"
            >
              <label
                class="rcx-box rcx-box--full rcx-check-box"
              >
                <input
                  aria-label="select frozen yoghurt"
                  checked=""
                  class="rcx-box rcx-box--full rcx-check-box__input"
                  type="checkbox"
                />
                <i
                  aria-hidden="true"
                  class="rcx-box rcx-box--full rcx-check-box__fake"
                />
              </label>
            </td>
            <th
              class="rcx-box rcx-box--full rcx-table__cell"
              scope="row"
            >
              Frozen yoghurt
            </th>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              159
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              6
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              24
            </td>
            <td
              class="rcx-box rcx-box--full rcx-table__cell--align-end rcx-table__cell"
            >
              4
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="rcx-box rcx-box--full rcx-table__selection rcx-css-15fugf6"
    >
      <div
        class="rcx-box rcx-box--full rcx-css-1vmb5x3"
      >
        5 Items selected
      </div>
      <div
        class="rcx-box rcx-box--full rcx-css-1tmdf8e"
      >
        <div
          class="rcx-button-group rcx-button-group--align-start rcx-css-x7bl3q"
          role="group"
        >
          <button
            class="rcx-box rcx-box--full rcx-button--small rcx-button rcx-button-group__item rcx-css-1yzvz7u"
            type="button"
          >
            <span
              class="rcx-button--content"
            >
              Delete
            </span>
          </button>
          <button
            class="rcx-box rcx-box--full rcx-button--small rcx-button rcx-button-group__item rcx-css-1yzvz7u"
            type="button"
          >
            <span
              class="rcx-button--content"
            >
              Cancel
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</body>
`;
